providers/tests/amazon/aws/hooks/test_mwaa.py

```diff
@@ -0,0 +1,130 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from unittest import mock
+
+import pytest
+from botocore.exceptions import ClientError
+from moto import mock_aws
+
+from airflow.providers.amazon.aws.hooks.mwaa import MwaaHook
+
+ENV_NAME = "test_env"
+PATH = "/dags/test_dag/dagRuns"
+METHOD = "POST"
+QUERY_PARAMS = {"limit": 30}
+
+
+class TestMwaaHook:
+    def setup_method(self):
+        self.hook = MwaaHook()
+
+    def test_init(self):
+        assert self.hook.client_type == "mwaa"
+
+    @mock_aws
+    def test_get_conn(self):
+        assert self.hook.get_conn() is not None
+
+    @pytest.mark.parametrize(
+        "body",
+        [
+            None,  # test case: empty body
+            {"conf": {}},  # test case: non-empty body
+        ],
+    )
+    @mock.patch.object(MwaaHook, "get_conn")
+    def test_invoke_rest_api_success(self, mock_conn, body) -> None:
+        boto_invoke_mock = mock.MagicMock(return_value=self.example_responses["success"])
+        mock_conn.return_value.invoke_rest_api = boto_invoke_mock
+
+        retval = self.hook.invoke_rest_api(ENV_NAME, PATH, METHOD, body, QUERY_PARAMS)
+        kwargs_to_assert = {
+            "Name": ENV_NAME,
+            "Path": PATH,
+            "Method": METHOD,
+            "Body": body if body else {},
+            "QueryParameters": QUERY_PARAMS,
+        }
+        boto_invoke_mock.assert_called_once_with(**kwargs_to_assert)
+        assert retval == {
+            k: v for k, v in self.example_responses["success"].items() if k != "ResponseMetadata"
+        }
+
+    @mock.patch.object(MwaaHook, "get_conn")
+    def test_invoke_rest_api_failure(self, mock_conn) -> None:
+        error = ClientError(
+            error_response=self.example_responses["failure"], operation_name="invoke_rest_api"
+        )
+        boto_invoke_mock = mock.MagicMock(side_effect=error)
+        mock_conn.return_value.invoke_rest_api = boto_invoke_mock
+        mock_log = mock.MagicMock()
+        self.hook.log.error = mock_log
+
+        with pytest.raises(ClientError) as caught_error:
+            self.hook.invoke_rest_api(ENV_NAME, PATH, METHOD)
+
+        assert caught_error.value == error
+        expected_log = {
+            k: v
+            for k, v in self.example_responses["failure"].items()
+            if k != "ResponseMetadata" and k != "Error"
+        }
+        mock_log.assert_called_once_with(expected_log)
+
+    @pytest.fixture(autouse=True)
+    def _setup_test_cases(self):
```

Comment:
Why not do this in the `setup_method` you already have instead of creating an autouse fixture?

Reply:
Also it's worth throwing a comment somewhere around this (wherever it ends up) mentioning that it is mutated by the code under test, which is why it's recreated before each test.

Reply:
Oh, I just realized, I incorrectly assumed that `setup_method` only runs once. Fixed now!